From stable-bounces@linux.kernel.org Fri Feb  8 04:30:29 2008
From: Jan Kara <jack@suse.cz>
Date: Fri, 08 Feb 2008 04:22:13 -0800
Subject: quota: turn quotas off when remounting read-only
To: torvalds@linux-foundation.org
Cc: akpm@linux-foundation.org, jack@suse.cz, stable@kernel.org
Message-ID: <200802081221.m18CLtGt024628@imap1.linux-foundation.org>


From: Jan Kara <jack@suse.cz>

patch 66191dc622f5ff0a541524c4e96fdacfacfda206 in mainline.

Turn off quotas before filesystem is remounted read only.  Otherwise quota
will try to write to read-only filesystem which does no good...  We could
also just refuse to remount ro when quota is enabled but turning quota off
is consistent with what we do on umount.

Signed-off-by: Jan Kara <jack@suse.cz>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 fs/super.c |    1 +
 1 file changed, 1 insertion(+)

--- a/fs/super.c
+++ b/fs/super.c
@@ -603,6 +603,7 @@ int do_remount_sb(struct super_block *sb
 			mark_files_ro(sb);
 		else if (!fs_may_remount_ro(sb))
 			return -EBUSY;
+		DQUOT_OFF(sb);
 	}
 
 	if (sb->s_op->remount_fs) {
